app/src/main/java/com/example/codechefeventsapp/fragments/HomeFragment.java

```diff
@@ -1,58 +1,115 @@
-        return inflater.inflate(R.layout.fragment_home, container, false);
+        View view = inflater.inflate(R.layout.fragment_home, container, false);
+        recyclerView=view.findViewById(R.id.pastRecyclerView);
+        recyclerView.setLayoutManager(new LinearLayoutManager(getContext()));
+        eventList=new ArrayList<>();
+        futureEvent=new ArrayList<>();
+        pastEvent=new ArrayList<>();
+        eventList.add(new Event("Coding", "", "JUL", "8", R.drawable.laptop));
+        eventList.add(new Event("Coding", "", "JUL", "25", R.drawable.laptop));
+        eventList.add(new Event("Coding", "", "APR", "25", R.drawable.laptop));
+        eventList.add(new Event("Coding", "", "JUL", "25", R.drawable.laptop));
+        eventList.add(new Event("Coding", "", "APR", "25", R.drawable.laptop));
+        eventList.add(new Event("Coding","Codechef","JUL","1",R.drawable.laptop));
+        eventList.add(new Event("Coding","Codechef","APR","27",R.drawable.laptop));
+        eventList.add(new Event("Coding","Codechef","APR","27",R.drawable.laptop));
+        eventList.add(new Event("Coding","Codechef","JUL","27",R.drawable.laptop));
+        eventList.add(new Event("Coding","Codechef","APR","27",R.drawable.laptop));
+        eventList.add(new Event("Coding","Codechef","APR","27",R.drawable.laptop));
+        eventList.add(new Event("Coding","Codechef","JUL","27",R.drawable.laptop));
+
+
+        Calendar c = Calendar.getInstance();
+        TimeZone time_zone = TimeZone.getTimeZone("IST");
+        c.setTimeZone(time_zone);
+        int day = c.get(Calendar.DATE);
+        int month = c.get(Calendar.MONTH);
+        month++;
+        day++;
+        for(int i=0;i<eventList.size();i++){
+            Event item=eventList.get(i);
+            String s= item.getEventDate();
+            int sDay=Integer.parseInt(s);
+            int sMonth = 0;
+            if(item.getEventMonth()=="JAN"){sMonth=1; }
+            else if(item.getEventMonth()=="FEB"){ sMonth=2; }
+            else if(item.getEventMonth()=="MAR"){ sMonth=3; }
+            else if(item.getEventMonth()=="APR"){ sMonth=4; }
+            else if(item.getEventMonth()=="MAY"){ sMonth=5; }
+            else if(item.getEventMonth()=="JUN"){ sMonth=6; }
+            else if(item.getEventMonth()=="JUL"){ sMonth=7; }
+            else if(item.getEventMonth()=="AUG"){ sMonth=8; }
+            else if(item.getEventMonth()=="SEPT"){sMonth=9; }
+            else if(item.getEventMonth()=="OCT"){sMonth=10; }
+            else if(item.getEventMonth()=="NOV"){sMonth=11; }
+            else if(item.getEventMonth()=="DEC"){sMonth=12; }
+            if(sMonth>month){
+                futureEvent.add(item);
+            }
+            else if(sMonth==month && sDay>=day){
+                futureEvent.add(item);
+            }
+            else{
+                pastEvent.add(item);
+            }
+        }
+
+        recyclerView.setAdapter(new pastEventAdapter(pastEvent));
```

Comment:
Always reference views after they have been created. So all this code has been shifted to `onViewCreated`